src/NServiceBus.Core/Features/FeatureActivator.cs

```diff
@@ -36,7 +36,7 @@ public void Add(Feature feature)
             }));
         }
 
-        public FeaturesReport SetupFeatures(IConfigureComponents container, PipelineSettings pipelineSettings)
+        public FeaturesReport SetupFeatures(IConfigureComponents container, PipelineSettings pipelineSettings, RoutingComponent routing)
```

Comment:
Those questions can't really be answered properly, as again, this is no proposal on how to write future components and how they need to be written. It's a proposal how to make routing available without the need to fiddle around with the ContextBag. If a second "component" turns out to work the same way, perfect, otherwise we can adjust.

> This would be a move we can't easily retract from because it adds to the public API surface.

coming up with more explicitly designed APIs will cause the same effect even more. Starting with a single API still allows us to easily retract from that API for other components without a breaking change. The downside would be inconsistency of course.

> What is a (preliminary) list of core components?

Transport, Serialization, Persistence are some we thought about.

> Can core components depend on one another? If so, how?

probably, we don't know yet. 
By requiring the other component as a dependency I guess.

> What is a lifecycle of a core component? When does it get initialized?

depends on the component I'd say? As long as it's API is usable where it is exposed, this 
shouldn't matter?

> Do core components have a uniform API or each one is unique like a snowflake?

currently they are unique like a snowflake. That allows to test API ideas on a small scale without having to refactor or align every single public API we expose.

>  If we pass IDispatcher to the Setup code, we can start sending messages there. Is it what we expect?

you could start sending raw transport messages, yes. Not sure this is really an issue. It would also allow us to execute the prestartup checks at a proper point instead of calling them in a feature startup task.